### PR TITLE
Make credential login a two-stage process

### PR DESCRIPTION
**Pull request by knolleary**
Part of #226 

This reworks the login screen to be a two stage process in preparation of support SSO flows.

Following the discussion in #226, I've implemented 'option 2':

1. First the user enters their username/email and clicks login

    image

2. They are then prompted for their password:
  
  image

I've added a workflow doc (copied inline below) to show the underlying http flows used - the SSO part is a bit of a placeholder, but shows where they will slot in place.

```mermaid
sequenceDiagram
participant U as User
participant B as Browser
participant RT as ForgeApp
participant DB as Database
participant IDP as IdentifyProvider
U->>B: Enters email on sign-up page
U->>B: Clicks login
B->>RT: POST /account/login (username=XYZ)
RT->DB: Checks username/email against list of SSO registered domains
alt Email not SSO enabled
    RT->>B: 403:{ code: 'password_required' }
    B->>U: Shows password box
    U->>B: Enters password
    U->>B: Clicks login
    B->>RT: POST /account/login (username/password)
    RT->>DB: Validates username/password
    RT->>B: 200:{}
end
alt Email SSO enabled
    RT->>B: 403:{ code:'sso_required', redirect:'/account/login/sso?u=<base64 encoded email>' }
    B->>RT: GET /account/login/sso?u=<base64 encoded email>
    RT->>RT: passport.authenticate
    RT->>IDP: SAML exchange
    IDP->>IDP: User authentication
    IDP->>RT: POST <TBD>
    RT->>DB: Verify Email against users
    alt Valid User
    RT->>DB: create session
    RT->>B: redirect to /
    else Unknown User
    RT->>B: redirect to /
    end
end
```

### CODE REVIEW

**Comment by knolleary** [comment]
Noting the failing cypress tests due to the recent upgrade of cypress that needs fixing in main. That will be masking failures in `auth.spec.js` - but without being able to run cypress locally currently, I can't fix those tests.

Once the cypress upgrade has been fixed in main, I'll rebase this branch and fix the tests.

**Comment by knolleary** [comment]
Rebased to pick up cypress fix and auth test updated for UI changes.

File: test/e2e/frontend/cypress/tests/auth.spec.js [hunk 1]
```diff
@@ -2,36 +2,72 @@ describe('FlowForge - Auth', () => {
...
+        cy.get('div[label=username] input').should('be.visible')
+        cy.get('div[label=password] input').should('not.exist')
+        // fill out username
         cy.get('div[label=username] input').type('alice')
+        // click "login"
+        cy.get('[data-action="login"]').click()
+        // should prompt for password as well
+        cy.get('div[label=username] input').should('be.visible')
+        cy.get('div[label=password] input').should('be.visible')
+        // fill out username
```
Comment (joepavitt) [inline test/e2e/frontend/cypress/tests/auth.spec.js:70]:
```suggestion
        // fill out pasword
```

**Review by joepavitt — COMMENTED** [review]
Minor comment correct

**Comment by knolleary** [comment]
Good spot on the enter handling.. missed that. 👍🏻

**Comment by knolleary** [comment]
@joepavitt are we good to merge?

**Review by joepavitt — APPROVED** [review]

**Pull request merged**